packages/dev/core/src/Lights/LTC/ltcTextureTool.ts

```diff
@@ -16,9 +16,14 @@ export interface IAreaLightLTCProvider {
     ltc1Texture: BaseTexture;
 
     /**
-     * Linearly trasnformed cossine texture Fresnel Approximation.
+     * Linearly transformed cosine texture Fresnel Approximation.
      */
     ltc2Texture: BaseTexture;
+
+    /**
+     * Promise to wait for Area Lights are ready.
+     */
+    whenAreaLightsReady: Promise<void>;

```

Comment:
This is not a great pattern. It's possible for someone to get an undefined ltc texture with the contract. Consider returning the ltc textures in the Promise result and rename function to `getTexturesAsync`.